src/coreclr/jit/emitriscv64.cpp

```diff
@@ -1627,7 +1627,11 @@ unsigned emitter::emitOutputCall(const insGroup* ig, BYTE* dst, instrDesc* id, c
         int reg2 = ((int)addr & 1) + 10;
         addr     = addr ^ 1;
 
-        assert(isValidSimm32(addr - (ssize_t)dst));
+        if (!emitComp->IsTargetAbi(CORINFO_NATIVEAOT_ABI))
```

Comment:
```suggestion
        if (!opts.compReloc)
```
I assume that this assert is invalid anytime we are emitting explicit relocs.